warehouse/integrations/clickhouse/clickhouse_test.go

```diff
@@ -36,210 +43,277 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestIntegrationClickHouse(t *testing.T) {
+func TestIntegration(t *testing.T) {
 	if os.Getenv("SLOW") != "1" {
 		t.Skip("Skipping tests. Add 'SLOW=1' env var to run test.")
 	}
 
-	var dbs []*sql.DB
-	for _, host := range []string{"wh-clickhouse", "wh-clickhouse01", "wh-clickhouse02", "wh-clickhouse03", "wh-clickhouse04"} {
-		ch := clickhouse.New()
-		db, err := ch.ConnectToClickhouse(clickhouse.Credentials{
-			Host:          host,
-			User:          "rudder",
-			Password:      "rudder-password",
-			DBName:        "rudderdb",
-			Secure:        "false",
-			SkipVerify:    "true",
-			TLSConfigName: "",
-			Port:          "9000",
-		}, true)
-		require.NoError(t, err)
+	c := testcompose.New(t, "testdata/docker-compose.yml")
 
-		err = db.Ping()
-		require.NoError(t, err)
+	t.Cleanup(func() {
+		c.Stop(context.Background())
+	})
+	c.Start(context.Background())
 
-		dbs = append(dbs, db)
-	}
+	misc.Init()
+	validations.Init()
+	warehouseutils.Init()
+	encoding.Init()
 
-	var (
-		provider = warehouseutils.CLICKHOUSE
-		jobsDB   = testhelper.SetUpJobsDB(t)
-		tables   = []string{"identifies", "users", "tracks", "product_track", "pages", "screens", "aliases", "groups"}
-	)
+	jobsDBPort := c.Port("jobsDb", 5432)
+	minioPort := c.Port("minio", 9000)
+	transformerPort := c.Port("transformer", 9090)
+	port := c.Port("clickhouse", 9000)
+	clusterPort1 := c.Port("clickhouse01", 9000)
+	clusterPort2 := c.Port("clickhouse02", 9000)
+	clusterPort3 := c.Port("clickhouse03", 9000)
+	clusterPort4 := c.Port("clickhouse04", 9000)
 
-	testCases := []struct {
-		name                        string
-		sourceID                    string
-		destinationID               string
-		writeKey                    string
-		warehouseEvents             testhelper.EventsCountMap
-		warehouseModifiedEvents     testhelper.EventsCountMap
-		clusterSetup                func(t testing.TB)
-		db                          *sql.DB
-		s3EngineEnabledWorkspaceIDs []string
-	}{
-		{
-			name:          "Single Setup",
-			sourceID:      "1wRvLmEnMOOxNM79pwaZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKph2Zn6XfTP7PI",
-			writeKey:      "C5AWX39IVUWSP2NcHciWvqZTa2N",
-			db:            dbs[0],
-		},
-		{
-			name:          "Single Setup with S3 Engine",
-			sourceID:      "1wRvLmEnMOOxNM79pwaZhyCqXRE",
-			destinationID: "21Ev6TI6emCFDKph2Zn6XfTP7PI",
-			writeKey:      "C5AWX39IVUWSP2NcHciWvqZTa2N",
-			db:            dbs[0],
-			s3EngineEnabledWorkspaceIDs: []string{
-				"BpLnfgDsc2WD8F2qNfHK5a84jjJ",
-			},
```

Comment:
This has been removed since now we are running the rudder server locally and accessing the minio and clickhouse using localhost, it is not possible to use the same host&port configured for load files generation and [loadByCopyCommand](https://github.com/rudderlabs/rudder-server/blob/e0280e4bf35323844fc0436a56f4803a739b8cd7/warehouse/integrations/clickhouse/clickhouse.go#L531) because clickhouse can't access minio using the same host&port.

Reply:
This has already been covered in a unit test. https://github.com/rudderlabs/rudder-server/blob/e0280e4bf35323844fc0436a56f4803a739b8cd7/warehouse/integrations/clickhouse/clickhouse_test.go#L341